src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacFieldAndReferencedFieldMismatch/DacFieldAndReferencedFieldMismatchAnalyzer.cs

```diff
@@ -0,0 +1,155 @@
+
+				CheckProperty(context, property, foreignReferenceAttribute!, dac, pxContext);
+			}
+		}
+
+		private static DacFieldAttributeInfo? TryGetForeignReferenceAttribute(PXContext pxContext, DacPropertyInfo property) =>
+			property.Attributes.FirstOrDefault(x => ContainsAttributeMatching(x, pxContext.AttributeTypes.PXSelectorAttribute.Type));
+
+		private void CheckProperty(SymbolAnalysisContext context, DacPropertyInfo property, DacFieldAttributeInfo foreignReferenceAttribute, 
+			DacSemanticModel dac, PXContext pxContext)
+		{
+			var localTypeAttribute = property.Attributes.FirstOrDefault(
+				x => ContainsAttributeMatching(x, pxContext.AttributeTypes.PXDBFieldAttribute)
+				|| ContainsAttributeMatchingOneOf(x, pxContext.AttributeTypes.UnboundDataTypeAttributes));
+
+			if (localTypeAttribute is null)
+				return;
+
+			var foreignField = ExtractForeignFieldSymbol(foreignReferenceAttribute, pxContext);
+
+			if (foreignField is not { ContainingSymbol: INamedTypeSymbol parentType })
+				return;
+
+			var foreignDac = GetOrInferModel(parentType, pxContext, cancellation: context.CancellationToken);
+
+			if (foreignDac?.PropertiesByNames.TryGetValue(foreignField.Name, out var foreignProperty) != true)
+				return;
+
+			if (HaveDifferentUnderlyingType(property, foreignProperty!))
+				ReportTypeMismatch(context, pxContext, localTypeAttribute, foreignProperty!.PropertyType.Name, property.Name, foreignField.Name);
+			else
+			{
+				var foreignTypeAttribute = foreignProperty!.Attributes.FirstOrDefault(
+					x => ContainsAttributeMatching(x, pxContext.AttributeTypes.PXDBFieldAttribute)
+					|| ContainsAttributeMatchingOneOf(x, pxContext.AttributeTypes.UnboundDataTypeAttributes));
+
+				// agreed to stop checking property here as the code may be incomplete at this point
+				if (foreignTypeAttribute is null)
+					return;
+
+				if (HaveDifferentFieldSize(localTypeAttribute, foreignTypeAttribute, pxContext))
+					ReportTypeSizeMismatch(
+						context, pxContext, localTypeAttribute, property.Name, foreignField.Name, GetFieldSize(foreignTypeAttribute, pxContext));
+			}
+		}
+
+		private DacSemanticModel? GetOrInferModel(INamedTypeSymbol type, PXContext pxContext, CancellationToken cancellation)
+		{
+			return _modelsCache.TryGetValue(type, out var model)
+				? model
+				: _modelsCache[type] = DacSemanticModel.InferModel(pxContext, type, cancellation: cancellation);
+		}
+
+		private static bool HaveDifferentUnderlyingType(DacPropertyInfo localProperty, DacPropertyInfo externalProperty) =>
+			!SymbolEqualityComparer.Default.Equals(localProperty.PropertyType, externalProperty.PropertyType);
+
+		private static ISymbol? ExtractForeignFieldSymbol(DacFieldAttributeInfo referenceAttribute, PXContext pxContext)
+		{
+			var argument = referenceAttribute.AttributeData.ConstructorArguments.FirstOrDefault();
+
+			if (argument.Value is INamedTypeSymbol { IsGenericType: true } symbol
+				&& symbol.Name == "Search")
```

Comment:
This check should be enhanced, there are two problems with it.

The main issue is that this check should account for potential custom BQL operators with the same functionality. This is a rare case, but it's not impossible, and our other BQL checks verify inheritance from some of the platform BQL types or implementation of BQL interfaces. In this case I think it would be correct to check for implementation of the `PX.Data.IBqlSearch` interface.

The second problem is that we don't usually use hard coded type names, especially plain type names. We have 
`Acuminator.Utilities.Roslyn.Constants.TypeFullNames` class for string constants. Usually, we add a new symbol to a suitable symbol set and use symbol comparison. If this is not possible, then we use full type name for comparison:
```cs
symbol.ToString() == "<Full Type Name>"
```

I would redo this check like this:
1. There is already a `IBqlSearch` constant in the `Acuminator.Utilities.Roslyn.Constants.TypeFullNames` class, so no need to add it again
2. There is no suitable symbol for the `IBqlSearch` constant, so I would add a new `IBqlSearch` symbol to `Acuminator.Utilities.Roslyn.Semantic.Symbols.BQLSymbols` symbol set:
```cs
public INamedTypeSymbol IBqlSearch => Compilation.GetTypeByMetadataName(TypeFullNames.IBqlSearch)!;
```
3. Find the suitable DAC field generic type arg by checking the type hierarchy for types that implement `IBqlSearch` going from base types to derived. I would do it with something like this:
```cs
public static INamedTypeSymbol? GetDacFieldFromBqlSearch(this INamedTypeSymbol bqlSearchType, PXContext pxContext)
{
	var hierarchyFromBaseToDerived = bqlSearchType.GetBaseTypesAndThis()
												  .TakeWhile(t => t.SpecialType != SpecialType.System_Object)
												  .OfType<INamedTypeSymbol>()
												  .Reverse();
	foreach (var type in hierarchyFromBaseToDerived)
	{
		if (type.ImplementsInterface(pxContext.BQL.IBqlSearch) && type.IsGenericType)
		{
			var dacBqlField = type.TypeArguments.FirstOrDefault(typeArg => typeArg.IsDacBqlField(pxContext)) as INamedTypeSymbol;

			if (dacBqlField != null)
				return dacBqlField;
		}
	}

	return null;
}
```